src/index.js

```diff
@@ -280,6 +284,7 @@ function createConfig(options, entry, format, writeMeta) {
 	}
 
 	const useTypescript = extname(entry) === '.ts' || extname(entry) === '.tsx';
+	const useReason = extname(entry) === '.re';
```

Comment:
is it unusual to mix `.js` & `.re`? I mean - maybe we should search whole source/src directory with glob to see if there is any `.re` in there?

Reply:
If it's unusual then we should only check the main entry point, like u did

Reply:
that'd be a nice trick - we wouldn't have to crawl for .re files at all. it's either a Reason project or it isnt.

Reply:
Well, I think Reason / BuckleScript is designed to run side by side with vanilla JS code.

What would the preferred approach be?

- .re as entry point => reason project
- make mixing possible?

Reply:
Kinda depends on how reason is uses (I had yet no chance to play with it, so cannot say from experience) - if mixing is popular we can allow it, if not then do it right now. We can also assume at the moment that we do not support mixing and revisit later if somebody requests it.

Reply:
Reason can definitely be mixed. My current use case is to slowly migrate a TypeScript project.